web/client/plugins/widgetbuilder/enhancers/layerSelector.js

```diff
@@ -11,6 +11,12 @@ import { compose, withState, mapPropsStream } from 'recompose';
 import { addSearch } from '../../../observables/wms';
 import { recordToLayer } from '../../../utils/CatalogUtils';
 
+// layers like tms or wmts don't need the recordToLayer convertion
+const toLayer = (r, service) => ["tms", "wmts"].includes(service.type) ? r : recordToLayer(r);
+
+// checks for tms wmts inorder to addSearch() to skip addSearch
+const addSearchObservable = (selected, service) => ["tms", "wmts"].includes(service.type) ? Rx.Observable.of(selected) : addSearch(recordToLayer(selected));
```

Comment:
I fixed a little bit this part. The only part is missing and you have to add is the remove of autkey param. See epics/catalog.js. It uses authkeyParamNameSelector to get and call recordToLayer and remove authkey from params. You have to do the same, and add this to your recordToLayer Call, as third parameter

```suggestion

// layers like tms or wmts don't need the recordToLayer conversion
const toLayer = (r, service) => ["tms", "wfs"].includes(service.type) // for tms and wfs the layer is ready
    ? r : recordToLayer(
        r, service.type === "wmts" // the type wms is default (for csw and wms), wmts have to be passed. // TODO: improve and centralize more
            ? "wmts"
            : undefined
    ); //

// checks for tms wmts in order to addSearch() to skip addSearch
const addSearchObservable = (selected, service) => ["tms", "wmts"].includes(service.type) ? Rx.Observable.of(toLayer(selected, service)) : addSearch(toLayer(selected, service));
```